cargo-dylint/tests/ci.rs

```diff
@@ -713,6 +715,103 @@ fn update() {
     });
 }
 
+#[test]
+fn lint_script_test() {
+    preserves_cleanliness("lint_script_test", false, || {
+        let cargo_dylint_path = Path::new("target/debug/cargo-dylint")
+            .canonicalize()
+            .expect("Failed to find cargo-dylint executable. Ensure it is built, e.g., with `cargo build -p cargo-dylint`.");
+
+        let example_restriction_dir = Path::new("examples/restriction");
+        let mut restriction_libs = Vec::new();
+        if example_restriction_dir.is_dir() {
+            for entry in read_dir(example_restriction_dir).unwrap() {
+                let entry = entry.unwrap();
+                if entry.path().is_dir() {
+                    let lib_name = entry.file_name().into_string().unwrap();
+                    // Exclude overscoped_allow as in the script
+                    if lib_name != "overscoped_allow" && lib_name != ".cargo" {
+                        restriction_libs.push(format!("--lib {lib_name}"));
+                    }
+                }
+            }
+        }
+        let restrictions_as_flags = restriction_libs.join(" ");
+
+        let mut experimental_dirs_str = Vec::new();
+        let example_experimental_dir = Path::new("examples/experimental");
+        if example_experimental_dir.is_dir() {
+            for entry in read_dir(example_experimental_dir).unwrap() {
+                let entry = entry.unwrap();
+                let path = entry.path();
+                if path.is_dir() && path.file_name().unwrap() != ".cargo" {
+                    // Convert path to string for the DIRS array
+                    experimental_dirs_str.push(path.to_string_lossy().to_string());
+                }
+            }
+        }
+
+        let base_flags =
+            format!("--lib general --lib supplementary {restrictions_as_flags} --lib clippy");
+
+        let mut dirs_to_lint = vec![
+            ".",
+            "driver",
+            "utils/linting",
+            "examples/general",
+            "examples/supplementary",
+            "examples/restriction",
+            "examples/testing/clippy",
+        ]
+        .into_iter()
+        .map(String::from)
```

Comment:
```suggestion
        .map(PathBuf::from)
```
Since these are really paths, let's please make them `PathBuf`s.